plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/plugins/index.tsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { SelectControl } from '@wordpress/components';
+import {
+	// @ts-expect-error no exported member.
+	PluginDocumentSettingPanel,
+} from '@wordpress/editor';
+/**
+ * Internal dependencies
+ */
+import useProductTypeSelector from '../hooks/use-product-type-selector';
+
+export const productTypeSelectorTitle = __( 'Product Type', 'woocommerce' );
+
+function ProductTypeSwitcher() {
+	const { productTypes, current, set } = useProductTypeSelector();
+
+	return (
+		<SelectControl
+			label={ __( 'Type switcher', 'woocommerce' ) }
+			value={ current?.slug }
+			options={ productTypes.map( ( productType ) => ( {
+				label: productType.label,
+				value: productType.slug,
+			} ) ) }
+			onChange={ ( slug ) => set( slug ) }
+			help={ __(
+				'Switch product type to see how the template adapts to each one.',
+				'woocommerce'
+			) }
+		/>
+	);
+}
+
+export default function PluginDocumentSettingTemplateSelectorPanel() {
```

Comment:
What do you think about making this name a bit more specific, something like `ProductTypeSelectorPlugin` or something like that? With the current name, it's not clear to me it allows changing the product type.